lib/pbench/server/api/resources/users_api.py

```diff
@@ -386,6 +407,76 @@ def get(self, username):
             )
             abort(403, message="Authentication token does not belong to the user")
 
+    @Auth.token_auth.login_required(f=Auth().verify_auth())
+    def put(self, username):
+        """
+        PUT request for updating user data.
+        This requires a Pbench auth token in the header field
+
+        This requires a JSON data with required user registration fields that needs an update
+        Example Json:
+        {
+            "first_name": "new_name",
+            "password": "password",
+            ...
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            Authorization:  Bearer Pbench_auth_token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "message": "Successfully registered."/"failure message",
+                    "data": {
+                        "username": <username>,
+                        "first_name": <firstName>,
+                        "last_name": <lastName>,
+                        "registered_on": registered_on,
+                    }
+                }
+        """
+        post_data = request.get_json()
+        try:
+            user, verified = self.auth.verify_user(username)
+        except Exception:
+            self.logger.exception("Exception occurred while verifying the user")
+            abort(500, message="INTERNAL ERROR")
+
+        # TODO: Check if the user has the right privileges
+        if verified or user.is_admin():
+            try:
+                # We will update the user object with the keys and values provided in the request payload.
+                # THe keys need to match with the column names in the user model. However, if any key in
+                # the payload does not match with the column name we just skip that field.
+                user.update(**post_data)
```

Comment:
Since this is direct user input, validating the keys passed in is probably a good idea before making the call.

Reply:
Yes I added a simple json check on the request payload since that's a hard requirement for the payload. Also logging if the user payload contain fields that are either non-updatabale or are not present in the user db. However, I am not removing the non existent field from json payload inside `PUT` method right away but rather just skipping them while updating the user object since we anyway have to iterate over all the fields from the payload